hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/BaseFileChecksumHelper.java

```diff
@@ -191,6 +207,35 @@ private FileChecksum makeMd5CrcResult() {
         crcPerBlock, fileMD5);
   }
 
+  FileChecksum makeCompositeCrcResult() throws IOException {
+    long blockSizeHint = 0;
+    if (keyLocationInfos.size() > 0) {
+      blockSizeHint = keyLocationInfos.get(0).getLength();
+    }
+    CrcComposer crcComposer =
+        CrcComposer.newCrcComposer(DataChecksum.Type.CRC32, blockSizeHint);
```

Comment:
I think, we need to use the `checksumData.getType()` from the ContainerProtos.

```
ContainerProtos.ChecksumData checksumData = chunkInfos.get(0).getChecksumData();
BaseFileChecksumHelper#setChecksumType(checksumData.getType());
```

Reply:
Yes this is part of the scope in HDDS-6127.